lib/solarwinds_apm/config.rb

```diff
@@ -151,6 +151,14 @@ def self.[]=(key, value)
       when :transaction_settings
         compile_settings(value)
 
+      when :trigger_tracing_mode
+        if key.instance_of?(String)
+          SolarWindsAPM.logger.warn {"[#{name}/#{__method__}] :trigger_trace must be in symbol e.g. :enabled or :disabled. Set to default :enabled"} 
+          @@config[key.to_sym] = :enabled
```

Comment:
Wondering if line 157 is needed--IIUC the option default values are already defined in the generator template file.

Reply:
I think it's still needed. If user modified the solarwinds_apm_config.rb file that set `SolarWindsAPM::Config[:trigger_tracing_mode] = String`, then we have to set the default back; since the generator template file will be parsed in this `self.[]=(key, value)` function.

Reply:
> that in line 155 it's checking whether the configuration key rather than value is a symbol... which doesn't seem correct?

It should be value, not key..

> And that in the :tracing_mode option handling below (line 165) the enforcement of configuration value stored as symbol is now removed, hmm.

I removed it in last commit, but it seems like it should have similar enforcement like `:trigger_tracing_mode`.

> Also noticing that env var override handling seems to be happening different places, for example SW_APM_TAG_SQL in line 168 of this method, while SW_APM_TRIGGER_TRACING_MODE in the otel_config.rb (line 44). Not sure if that means SW_APM_TAG_SQL env var would only work if the customer also set :tag_sql ?

I should have consolidate them into one place. I was thinking to have one config file rather than two configs as well for simplicity.

> This is inherited code that doesn't seem to have much test coverage

I will create test coverage for config.rb

Reply:
> Not sure if that means SW_APM_TAG_SQL env var would only work if the customer also set :tag_sql ?

`SW_APM_TAG_SQL` will always work since the config will load the default template at first place no matter if user have config file or not. Default template will be loaded [here](https://github.com/solarwinds/apm-ruby/blob/main/lib/solarwinds_apm/config.rb#L104).